### Introduction to Probability

#### Probability is not possibility

Possibility is whether or not an event can occur, and is binary - True if it can, False if it cannot.  The possible events in a coin flip are two.  The possible events in a card draw? 52.

Probability, on the other hand, is the likelihood that a possible event occurs, given a random opportunity.  The probability of pulling a Jack from a deck is 4/52, or 0.0769; the probability of a coin landing on heads? 1/2 or 0.5.

#### The bounds of probability

The probability of an event occurring is always a fraction, or a number between 0 and 1.  When probability is 0, there is no chance a thing will happen.  When it is 1, it is certain to happen.  If there are 4 marbles in a bag, and all of them are blue, then the probability of a blue marble being drawn is 4/4, or 1 - a certainty.  

#### Additive probability

The probability of one **or** another event occurring is the **sum** of the individual probabilities, if the events are mutually exclusive and cannot co-occur.

What if they **can** co-occur?  Then, to get P(one or the other) one must sum P(one) and P(other), then **subtract** P(one & the other).

#### Multiplicative probability

The probability of two events **both** occurring is the **product** of the individual probabilities, if the events are independent.

#### Conditional probability - Bayes Rule

If two events are NOT independent, what we have is **conditional probability**.  Bayes Rule, dealing with conditional probability, states that if we know the independent rates of occurrence of two events, we can find the probability of one event given that the other has occurred.

*Probability of Disease Given Symptom = Probability of Disease * Probability of Symptom, Given Disease / Probability of Symptom*



In [1]:
# Additive Probability

# Liver Disease Example, page 52
total_patients = 20
prob_hepB = 4/20
prob_cirrhosis = 6/20
prob_both = 2/20

# If a patient is selected at random, what is the probability of Hepatitis B OR Cirrhosis being the diagnosis?
prob_hepB + prob_cirrhosis - prob_both

0.4

In [2]:
# Multiplicative Probability

# Probability of Name Draws, p. 52
prob_Y = 0.4
prob_M = 0.2

prob_Y_and_M = prob_Y * prob_M
print("The probability of drawing both a name starting with 'Y' and a name starting with 'M' is " + str(prob_Y_and_M))

The probability of drawing both a name starting with 'Y' and a name starting with 'M' is 0.08000000000000002


## Week 3: Probability Applied

In [3]:
# Conditional Probability, Independent Events: The probability of cirrhosis, given hepatitis B

# Probability of both hep B & cirrhosis, given hep B:

prob_both / prob_hepB

0.5

In [4]:
# Conditional Probability, Dependent Events: The probability of both hepatitis B and cirrhosis (20 patients)

prob_hepB * (prob_both/prob_hepB)

0.1

In [5]:
# Conditional Probability, Dependent Events: The probability of both hepatitis B and cirrhosis (25 patients), p. 53

4/25 * ((2/25)/(4/25))

0.08

In [6]:
# Bayes Rule: the probability that a person with fever has the flu, p. 53

prob_fever = 0.06
prob_flu = 0.03
prob_fever_given_flu = 0.9

# Probability that a person with fever has flu:
(prob_flu * prob_fever_given_flu) / prob_fever

0.45

Based on these probabilities, there is a 45% chance a person with a fever has the flu.

### Probability in Medicine -  Assessing the Likelihood of an Increase in Post-Operative Length of Stay

Bayes Theorem can help to calculate the likelihood of an extended length of stay for a particular patient.  

For example, if the baseline likelihood of an extended length of stay after surgery is 10%, there are still a series of additional things that could happen to a patient after surgery that might increase the likelihood of an extended length of stay.  
For example, what if a post-operative infection could increase the likelihood of an extended stay by 75%, and >300 ml blood loss during surgery could increase it by 25%, and prolonged PACU time could increase it by 50%?

If one or more of these things happened to a patient after surgery, we could use Bayes Theorem to calculate how much more likely an extended stay would become!

#### Odds of an Increased Length of Stay - Baseline Definitions and Assumptions

1. H (Hypothesis) = Increased Length of Stay
2. Baseline Probability of H (Probability of Increased Length of Stay with NO additional events) = .1 (10%)
3. Therefore, Baseline Probability of not H (Probability of NOT having an Increased Length of Stay) = 1-.1 = .9 (90%)
4. Probability of a post-operative infection, 0.21% or 0.0021; P(post-op infection (POI), given H) = .75 (75%)

Reference: Zabaglo M, Sharman T. Postoperative Wound Infection. [Updated 2023 Jul 3]. In: StatPearls [Internet]. Treasure Island (FL): StatPearls Publishing; 2024 Jan-. Available from: https://www.ncbi.nlm.nih.gov/books/NBK560533/
5. Probability of surgical blood-loss > 300 milliliters, 0.07 (7%); P(EBL > 300 mls, given H) = .25 (25%)
6. Probability of PACU time > 2 hours, 0.2 (20%); P(PACU time > 2 hours, given H) = .5 (50%)


In [7]:
prob_H = 0.1
prob_Not_H = 1 - 0.1

prob_POI = 0.0021
prob_H_given_POI = 0.75

prob_blood_loss = 0.07
prob_H_given_blood_loss = 0.25


prob_pacu_increase = 0.2
prob_H_given_pacu_increase = 0.5

#### We update our belief about the probability of an event given new evidence about a situation that may result in that event.  

In this case, we update our belief about the probability of an Increased Length of Stay for a post-operative patient given the occurrence of one or more adverse events!


In [8]:
incr_in_likelihood_of_H_over_baseline = prob_H * prob_H_given_POI / prob_POI
incr_in_likelihood_of_H_over_baseline

35.71428571428572

#### Adjusting the Probability of Increased LOS After Post-Op Infection

The probability of an Increased Length of Stay (H), given a Post-Operative Infection:  P(H | POI) = .10 (The baseline probability of an Increased Length of Stay) * .75 (The probability of Increased Length of Stay given Post-Op Infection) = 0.075 / 0.0021 (Probability of Post-Op Infection, .21%), results in an increase of 35.7% in the likelihood of an Increased Length of Stay over the baseline (0.10). **If POI occurs, the likelihood of an increased LOS is now 35.8%.**

In [10]:
prob_H + incr_in_likelihood_of_H_over_baseline

35.814285714285724

#### Adjusting the Probability of Increased LOS After EBL > 300 mls

The probability an Increased Length of Stay (H), given EBL > 300 mls: P(H | EBL > 300mls) = .10 * .25 = .025/ 0.07 (assuming the Probability of EBL > 300mls is 7%) = 0.357 or **increases probability of an increased LOS to 45.7%.**

In [11]:
incr_in_likelihood_of_H_over_baseline2 = prob_H * prob_H_given_blood_loss / prob_blood_loss
incr_in_likelihood_of_H_over_baseline2 + prob_H

0.4571428571428572

#### Adjusting the Probability of an Increased LOS After PACU time > 2 Hours

The probability of an Increased Length of Stay (H), given PACU time > 2 hours: P(H | PACU time > 2 hours) = .10 * .50 = 0.05/ 0.2 (assuming the Probability of PACU time > 2 hours is 0.2 or 20%) = 0.25.  **Increased PACU time increases the probability of an increased LOS to 35% or results in a 15% increase in the likelihood of an Increased Length of Stay over the baseline (0.10).**

In [12]:
incr_in_likelihood_of_H_over_baseline3 = prob_H * prob_H_given_pacu_increase / prob_pacu_increase
incr_in_likelihood_of_H_over_baseline3 + prob_H

0.35

### But what if all 3 things happen?  

In this case, we must accumulate the probability of an increased length of stay from one event to the next as time goes on!


The first thing that would impact the baseline P(LOS) of 10% would be an increase in surgical blood-loss. So, the new baseline P(H) would be 45.7%

In [13]:
prob_H2 = prob_H + incr_in_likelihood_of_H_over_baseline2
prob_H2

0.4571428571428572

The second thing that could impact LOS would be increased PACU time, so if that *also* occurred, we compute P(H2 | PACU time > 2 hours) with a baseline of 45.7% and get: .457 * .50 = 0.2285 / .2 = an additional 114% likelihood, and now baseline P(H) is 124%.  The chance of an increased Length of Stay is now > 100% and would be very likely.

**NOTE: While it is not possible for the probability of one action / situation to be > 100%, it *is* possible for cummulative probability rise above 100%.**

In [15]:
prob_H3 = prob_H2 * prob_H_given_pacu_increase / prob_pacu_increase
prob_H3 = prob_H3 + prob_H
prob_H3

1.2428571428571429

The last thing that could happen would be the post-op infection.  If we start with the latest baseline P(H) of 124%,  then 1.24 * .75 = 0.932 / 0.0021 = a final 443.88% probability of an increased LOS after all 3 events, and a **whopping 444% increase in the likelihood of an increased LOS over the original baseline probability of 10%!!**

In [23]:
prob_H4 = prob_H3 * prob_H_given_POI / prob_POI
prob_H4_post = prob_H4 + prob_H
print(prob_H4)
print(prob_H4_post)

443.8775510204082
443.9775510204082


#### Consider

**What do you think nurses or providers could do to prevent the increase in likelihood of an increased length-of-stay after surgery illustrated in this problem?**

**Given hospitals are penalized for extended patient lengths of stay, how important would this kind of analysis and the development of related interventions be to patients, staff, and business quality?**

#### References

Koehrsen, W. (2018, Feb 14).  Bayes’ Rule Applied.  Towards Data Science.  Retrieved from https://towardsdatascience.com/bayes-rule-applied-75965e4482ff

Riffenburg, R. H. & Gillen, D. L. (2020).  Statistics in Medicine, Fourth Edition.  Cambridge, MA: Elsevier.

Zabaglo M, Sharman T. Postoperative Wound Infection. [Updated 2023 Jul 3]. In: StatPearls [Internet]. Treasure Island (FL): StatPearls Publishing; 2024 Jan-. Available from: https://www.ncbi.nlm.nih.gov/books/NBK560533/